src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamedNested.java

```diff
@@ -0,0 +1,117 @@
+/*
+UnusedCatchParameterShouldBeUnnamed
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedcatchparametershouldbeunnamed;
+
+public class InputUnusedCatchParameterShouldBeUnnamedNested {
+
+
+    void testNestedCatch1() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) { // violation
+            try {
+                int y = 1 / 0;
+            } catch (Exception ex) {  // violation
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedCatch2() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) {
+            try {
+                int y = 1 / 0;
+            } catch (Exception ex) {  // violation
+                e.printStackTrace();
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedCatch3() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) { // violation
+            try {
+                int y = 1 / 0;
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedCatch4() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) {
+            try {
+                int y = 1 / 0;
+                e.printStackTrace();
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedCatch5() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception _) {
+            try {
+                int y = 1 / 0;
+            } catch (Exception _) {
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedCatchTwoLevels() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) {
+            try {
+                int y = 1 / 0;
+            } catch (Exception ex) { // violation
+                try {
+                    int z = 1 / 0;
+                } catch (Exception exx) {
+                    exx.printStackTrace();
+                    System.out.println("infinity");
+                }
+                e.printStackTrace();
+                System.out.println("infinity");
+            }
+            System.out.println("infinity");
+        }
+    }
+
+    void testNestedInsideAnonClass() {
+        try {
+            int x = 1 / 0;
+        } catch (Exception e) { // violation
```

Comment:
One more variation of this:
`````
    void testNestedInsideAnonClass() {
        try {
            int x = 1 / 0;
        } catch (final Exception e) { // should not be violation
            Runnable runnable = new Runnable() {
                @Override
                public void run() {
                    try {
                        int x = 1 / 0;
                    } catch (Exception ex) {
                        System.out.println(e.getMessage());
                    }
                }
            };
        }
    }
````

Reply:
done with the four variations :).
moved to separate input file

Reply:
Point to the file and lines please.

Reply:
Point to file and lines please

Reply:
https://github.com/checkstyle/checkstyle/blob/4867842d1e57e1605138eea29e109e99a601c00c/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamedInsideAnonClass.java

This file has all four variations from your example. Please let me know if I misunderstood you. You meant variation when the inner exception is used/outer is not used and vise versa right?

Reply:
Relook at my example here. Top catch and inner catch have different catch parameter names, and inner get message uses top catch's name. The examples I see in the input file you showed are all the same variable name.

Reply:
Yeah my bad I didn't notice the different names. Done 
https://github.com/checkstyle/checkstyle/blob/66e0f2a3c2cf19a4849ac2f4c714e2590519f450/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unusedcatchparametershouldbeunnamed/InputUnusedCatchParameterShouldBeUnnamedInsideAnonClass.java#L83-L113

Is there any extra confusions ? :)

Reply:
Nope. I think I am good.